TYPO3.Flow/Classes/TYPO3/Flow/Reflection/ObjectAccess.php

```diff
@@ -245,6 +246,30 @@ public static function setProperty(&$subject, $propertyName, $propertyValue, $fo
             } else {
                 $subject->$propertyName = $propertyValue;
             }
+        } elseif (self::isCollectionTypePropertyValue($propertyValue)
+                  && is_callable(array($subject, $addMethodName = self::buildAdderMethodName($propertyName)))
+                  && is_callable(array($subject, $removeMethodName = self::buildRemoverMethodName($propertyName)))) {
+            $itemsToAdd = is_object($propertyValue)?iterator_to_array($propertyValue):(array)$propertyValue;
```

Comment:
It's not explicitly stated IIRC, but the [examples](https://github.com/php-fig-rectified/fig-rectified-standards/blob/master/PSR-2-R-coding-style-guide-additions.md#ternary-operator) suggest the same!